rfcs/0042-config-option.md

```diff
@@ -0,0 +1,242 @@
+
+### Valuable options
+
+| Kind | Why | Examples | Notes |
+| --- | --- | --- | --- |
+| Main/popular settings | These options are what you'll need for basic module usage, they provide a good overview and should be enough for most users | [`services.i2pd.address`](https://nixos.org/nixos/manual/options.html#opt-services.i2pd.address), [`services.taskserver.organisations`](https://nixos.org/nixos/manual/options.html#opt-services.taskserver.organisations) but **not** [~~`services.i2pd.logCLFTime`~~](https://nixos.org/nixos/manual/options.html#opt-services.i2pd.logCLFTime) and **not** [~~`services.taskserver.extensions`~~](https://nixos.org/nixos/manual/options.html#opt-services.taskserver.extensions) | Settings only needed by few can be set through the `settings` option instead |
+| Mandatory user-specific values | Reminds the user that they have to set this in order for the program to work, an evaluation error will catch a missing value early | [`services.hydra.hydraURL`](https://nixos.org/nixos/manual/options.html#opt-services.hydra.hydraURL), [`services.davmail.url`](https://nixos.org/nixos/manual/options.html#opt-services.davmail.url) | |
+| Sensitive data, passwords | To avoid those ending in the Nix store, ideally an option like `passwordFile` should replace a password placeholder in the configuration file at runtime | | This is specifically about configuration files that have a `password`-like setting |
+
+## Backwards compatibility with existing modules
+
+This RFC has to be thought of as a basis for *new* modules first and foremost. By using this approach we can provide a good basis for a new module, with great flexibility for future changes.
+
+For existing modules, it is often not possible to use this `settings` style without breaking backwards compatibility. How this is handles is left up to the module authors. A workaround that could be employed is to define options `useLegacyConfig` or `declarative` which determin  the modules behavior in regards to old options.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+For [Part 2][part2]:
+- The less encoded options there are, the less checks are happening at evaluation time, and by default this means more runtime failures for initial runs, which isn't as bad as it sounds. If available, [configuration checking tools](#configuration-checking-tools) can be used to have build-time failures instead, or alternatively [assertions][assertions] can be used to have additional evaluation-time checks.
```

Comment:
Broken link #configuration-checking-tools

Reply:
Ohh I fixed this locally but forgot to push it, done now!